pkg/codegen/java/gen_test.go

```diff
@@ -224,10 +242,20 @@ func TestGeneratePackage(t *testing.T) {
 			) (map[string][]byte, error) {
 				pkg.Description = "test description"
 				pkg.Repository = "https://github.com/pulumi/pulumi-java"
-				pkg.Language = map[string]interface{}{
-					"java": testCase.packageInfo,
+
+				if pkg.Language == nil {
+					pkg.Language = map[string]interface{}{}
 				}
-				return GeneratePackage(tool, pkg, extraFiles, nil, false)
+				if pkg.Language["java"] == nil {
+					pkg.Language["java"] = testCase.packageInfo
+				} else {
+					if err := pkg.ImportLanguages(map[string]schema.Language{"java": Importer}); err != nil {
+						panic(err)
+					}
+					info := pkg.Language["java"].(PackageInfo)
+					pkg.Language["java"] = info.With(testCase.packageInfo)
+				}
+				return GeneratePackage(tool, pkg, extraFiles, nil, false /*local*/, false /*legacyBuildFiles*/)
```

Comment:
I do wonder if it's possible to find a way to test this set to `true`, since that's the behaviour we want to preserve, but equally it's probably fine?

Reply:
I updated `TestGenGradleProject` to check the generated file for both legacy falls and true variants.